src/components/ScatterPlot/ScatterPlot.jsx

```diff
@@ -0,0 +1,226 @@
+ * @prop {Number} height The height of the chart
+ * @prop {Function} onHover Callback for when a point is hovered on
+ * @prop {Number} width The width of the chart
+ * @prop {Array} yExtent Optional. The min and max value of the yKey in the chart
+ * @prop {String} yKey="y" The key in the data points to read the y value from
+ * @prop {Array} xExtent Optional. The min and max value of the xKey in the chart
+ * @prop {String} xKey="x" The key in the data points to read the x value from
+ */
+export default class ScatterPlot extends PureComponent {
+  static propTypes = {
+    data: PropTypes.array,
+    height: PropTypes.number,
+    id: React.PropTypes.string,
+    onHover: PropTypes.func,
+    width: PropTypes.number,
+    xExtent: PropTypes.array,
+    xKey: PropTypes.string,
+    yExtent: PropTypes.array,
+    yKey: PropTypes.string,
+  }
+
+  static defaultProps = {
+    data: [],
+    yKey: 'y',
+    xKey: 'x',
+    width: 200,
+    height: 200,
+  }
+
+  /**
+   * Initiailize the vis components when the component is about to mount
+   */
+  componentWillMount() {
+    // Holds refs to chart node for line updating.
+    this.root = null;
+
+    this.visComponents = this.makeVisComponents(this.props);
+  }
+
+  /**
+   * When the react component mounts, setup the d3 vis
+   */
+  componentDidMount() {
+    this.setup();
+  }
+
+  /**
+   * When new component is updating, regenerate vis components if necessary
+   */
+  componentWillUpdate(nextProps) {
+    // regenerate the vis components if the relevant props change
+    this.visComponents = this.makeVisComponents(nextProps);
+  }
+
+  /**
+   * When the react component updates, update the d3 vis
+   */
+  componentDidUpdate() {
+    this.update();
+  }
+
+  /**
+   * Initialize the d3 chart - this is run once on mount
+   */
+  setup() {
+    this.g = d3.select(this.root)
+      .append('g'); // transformed to have margin in update()
+
+    // add in axis groups
+    this.xAxis = this.g.append('g').classed('x-axis', true);
+    this.yAxis = this.g.append('g').classed('y-axis', true);
+
+    this.update();
+  }
+
+  /**
+   * Update the d3 chart - this is the main drawing function
+   */
+  update() {
+    this.updateAxes();
+    this.updateChart();
+  }
+
+  /**
+   * Iterates through data, using line generators to update charts.
+   */
+  updateChart() {
+    const {
+            data,
+            innerMargin,
+            colors,
+            xKey,
+            xScale,
+            yKey,
+            yScale,
+          } = this.visComponents;
+
+    this.g.attr('transform', `translate(${innerMargin.left} ${innerMargin.top})`);
+
+    const binding = this.g.selectAll('.data-point').data(data, d => d.id);
+    binding.exit().remove();
+    const entering = binding.enter().append('circle')
+      .classed('data-point', true);
+
+    // TODO: the values inside our summary data don't all get populated at once?
+    binding.merge(entering)
```

Comment:
It's the same as time series and hourly in that we get the city (last week, last month, last year) in one call, and each ISP (last week, last month, last year) in one call per ISP.